src/cart/crt_context.c

```diff
@@ -1213,14 +1213,13 @@ crt_context_timeout_check(struct crt_context *crt_ctx)
 		crt_req_timeout_untrack(rpc_priv);
 		rpc_priv->crp_timeout_ts = 0;
 
-		d_list_add_tail(&rpc_priv->crp_tmp_link, &timeout_list);
+		d_list_add_tail(&rpc_priv->crp_tmp_link_timeout, &timeout_list);
```

Comment:
[Not request for changes] I suggest we assert `d_list_empty(&rpc_priv->crp_tmp_link_timeout)`.